client/src/components/Search/SearchBox.js

```diff
@@ -12,131 +12,136 @@ import TagPropType from '../Tag/TagPropType';
-    this.componentDidUpdate = this.componentDidUpdate.bind(this);
-    this.onResize = this.onResize.bind(this);
-    this.setWidth = this.setWidth.bind(this);
-    this.renderTags = this.renderTags.bind(this);
-    this.getComponentWidth = this.getComponentWidth.bind(this);
-    this.calculateSpaceForTags = this.calculateSpaceForTags.bind(this);
-    this.calculateInputLeftPadding = this.calculateInputLeftPadding.bind(this);
-    this.calculateInputRightPadding = this.calculateInputRightPadding.bind(this);
-    this.onTagListResize = this.onTagListResize.bind(this);
-    this.focusOnLastTag = this.focusOnLastTag.bind(this);
-    this.focusOnInput = this.focusOnInput.bind(this);
-
-    this.state = {
-      hasFocus: false,
-      width: window.innerWidth - 180 - 55,
-      tagWidth: 0
-    };
-  }
+const SearchBox = ({
+  onSearch,
+  onToggleFilter,
+  onHideFilter,
+  onChange,
+  onHide,
+  onTagDelete,
+  onTagClick,
+  placeholder = i18n._t('Admin.SEARCH', 'Search'),
+  expanded,
+  formId,
+  id,
+  searchText,
+  hideable,
+  showFilters,
+  name,
+  dirty,
+  clearable,
+  tagData = [],
+  borders = {
+    top: false,
+    right: false,
+    bottom: true,
+    left: true,
+  },
+  children
+}) => {
+  const [hasFocus, setHasFocus] = useState(false);
+  const [width, setWidthState] = useState(window.innerWidth - 180 - 55);
+  const [tagWidth, setTagWidth] = useState(0);
+  const nodeRef = useRef(null);
+  const isFirstRender = useRef(true);
+
+  const getComponentWidth = () => {
+    if (!nodeRef.current) {
+      return 0;
+    }
+    return nodeRef.current.getBoundingClientRect().width;
+  };
 
-  componentDidUpdate() {
-    const width = this.getComponentWidth();
-    this.setWidth(width);
-  }
+  const setWidth = (newWidth) => {
+    setWidthState((prevWidth) => {
+      if (prevWidth !== newWidth) {
+        return newWidth;
+      }
+      return prevWidth;
+    });
+  };
 
   /**
    * Detect resizing of the component.
    * If it's bigger, the summary view is triggered.
    * @param tagListDimension
    */
-  onResize(dimension) {
-    this.setWidth(dimension.width);
-  }
-
-  onTagListResize(dimensions) {
-    const tagWidth = dimensions.width;
-    if (this.state.tagWidth !== tagWidth) {
-      this.setState({ tagWidth });
-    }
-  }
-
-  setWidth(width) {
-    if (this.state.width !== width) {
-      this.setState({ width });
-    }
-  }
-
-  /**
-   * Measure the width of the component.
-   * @returns number
-   */
-  getComponentWidth() {
-    const node = this.nodeRef;
-    if (!node) {
-      return 0;
-    }
-
-    return node.getBoundingClientRect().width;
-  }
+  const onResize = (dimension) => {
+    setWidth(dimension.width);
+  };
+
+  const onTagListResize = (dimensions) => {
+    const newTagWidth = dimensions.width;
+    setTagWidth((prevTagWidth) => {
+      if (prevTagWidth !== newTagWidth) {
+        return newTagWidth;
+      }
+      return prevTagWidth;
+    });
```

Comment:
```suggestion
    setTagWidth(newTagWidth);
```
Won't this end up being the exact same result?

Reply:
Yup, updated